src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/cjsm/CjsmService.kt

```diff
@@ -2,14 +2,48 @@ package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.cjsm
 
 import com.amazonaws.services.s3.AmazonS3
 import org.springframework.stereotype.Service
+import java.io.BufferedInputStream
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.InputStreamReader
 
 @Service
-class CjsmService(private val amazonS3Client: AmazonS3, private val s3Config: S3Config) {
+class CjsmService(
+  private val amazonS3: AmazonS3,
+  private val s3Config: S3Config,
+  private val cjsmDirectoryRepository: CjsmDirectoryRepository
+) {
 
   fun readCjsmOrgs(): String =
     try {
-      amazonS3Client.getObjectAsString(s3Config.bucketName, s3Config.cjsmOrgsCsv)
+      amazonS3.getObjectAsString(s3Config.bucketName, s3Config.cjsmOrgsCsv)
     } catch (ex: Exception) {
       "An error occurred attempting to read ${s3Config.cjsmOrgsCsv} from S3 bucket ${s3Config.bucketName}: ${ex.message}"
     }
+
+  fun saveCjsmDirectoryCsv() =
+    amazonS3.getObject(s3Config.bucketName, s3Config.cjsmOrgsCsv)
+      .objectContent
+      .let { streamCjsmDirectoryCsv(it) }
```

Comment:
This isn't called yet - next job is to write a test for this that uses LocalStack and a large csv file, then wire it up to an endpoint.